gxa/src/main/java/uk/ac/ebi/atlas/controllers/rest/DifferentialJsonResultsParser.java

```diff
@@ -31,63 +32,69 @@ public DifferentialJsonResultsParser(ContrastTrader contrastTrader)
         }
     }
 
-    private DiffAnalytics buildDiffAnalyticObject(JsonElement e) {
+    private DiffAnalytics buildDiffAnalyticsObject(JsonElement e) {
 
         try {
 
             JsonObject resultObj = e.getAsJsonObject();
 
-            Contrast contrast = contrastTrader.getContrast(resultObj.get("experiment_accession").getAsString(), resultObj.get("contrast_id").getAsString());
+            String experiment_accession = resultObj.get("experiment_accession").getAsString();
+
+            String bioentity_identifier = resultObj.get("bioentity_identifier").getAsString();
+
+            String contrast_id = resultObj.get("contrast_id").getAsString();
+
+            Double p_value = resultObj.get("p_value").getAsDouble();
+
+            Double fold_change = resultObj.get("fold_change").getAsDouble();
+
+            String keyword_symbol = Optional.of(resultObj.get("keyword_symbol")).map(JsonElement::getAsString).orElse(bioentity_identifier);
+
+            JsonElement nullableTStatistic = resultObj.get("t_statistic");
 
-            DifferentialExpression expression =
-                    buildDifferentialExpression(
-                            resultObj.get("p_value").getAsDouble(),
-                            resultObj.get("fold_change").getAsDouble(),
-                            Optional.ofNullable(resultObj.get("t_statistic")).map(JsonElement::getAsDouble).orElse(null));
+            Optional<Double> tStatistic = nullableTStatistic == null ? Optional.empty() : Optional.of(nullableTStatistic.getAsDouble());
+
+            Function<Double, DifferentialExpression> microarrayExpressionMapper = ts -> new MicroarrayExpression(p_value, fold_change, ts);
+            DifferentialExpression expression = tStatistic.map(microarrayExpressionMapper).orElse(new DifferentialExpression(p_value, fold_change));
+
+            Contrast contrast = contrastTrader.getContrast(experiment_accession, contrast_id);
+
+
+            /*  Data coming from database has species name following Binomial nomenclature notation for naming species i.e. first character of species to be capital
+                but data in solr doesn't follow this notation so to make it consistent first character of species is uppercase
+            */
 
             String species = resultObj.get("species").getAsString();
             species = species.substring(0,1).toUpperCase() + species.substring(1);
 
-
-            return new DiffAnalytics(resultObj.get("bioentity_identifier").getAsString(), Optional.ofNullable(resultObj.get("keyword_symbol")).map(JsonElement::getAsString).orElse(resultObj.get("bioentity_identifier").getAsString()), resultObj.get("experiment_accession").getAsString(), expression, species, contrast);
+            return new DiffAnalytics(bioentity_identifier, keyword_symbol, experiment_accession, expression, species, contrast);
         }
-        catch(RuntimeException exp)
-        {
+        catch(NullPointerException exp) {
```

Comment:
Which method is likely to throw a NPE? This is a down-to-earth, honest question.